weave/engine_trace.py

```diff
@@ -17,10 +17,32 @@
 import time
 import json
 import dataclasses
+from typing import Optional
 
 from . import logs
 from . import stream_data_interfaces
 
+from ddtrace import span as ddtrace_span
+
+# replaces ddtrace.Span.set_tag and ddtrace.Span.set_metric
+old_set_tag = ddtrace_span.Span.set_tag
+old_set_metric = ddtrace_span.Span.set_metric
+
+
+def set_tag(self, key, val="", pii_val=""):
```

Comment:
this is not defensive. It makes it the default / easy case that the dev accidentally logs the pii val. the first param should be non redacted, the second redacted.

Reply:
ah my mistake, thats what I intended, that the first val is non-redacted, my variable names did not make this clear, let me fix that